docker_cli/_shell_cmds.py

```diff
@@ -0,0 +1,369 @@
+        retval.extend(targets)
+        if clean:
+            retval += ["&&", "apt-get", "clean", "all"]
+        if stringify:
+            return " ".join(retval)
+        return retval  # pragma: no cover
+
+    @staticmethod
+    def generate_package_command(  # pragma: no cover
+        target: str, *, stringify: bool = False
+    ) -> Union[List[str], str]:
+        retval = ["dpkg", "-i", target]
+        if stringify:
+            return " ".join(retval)
+        return retval
+
+    @staticmethod
+    def generate_configure_command(*, stringify: bool = False) -> Union[List[str], str]:
+        retval = []
+        retval += ["apt-get", "-y", "update"]
+        retval += ["&&", "apt-get", "clean", "all"]
+        if stringify:
+            return " ".join(retval)
+        return retval  # pragma: no cover
+
+    @property
+    def name(self):
+        return "apt-get"
+
+    @property
+    def file_type(self):  # pragma: no cover
+        return "deb"
+
+
+class URLReader(ABC):
+    """
+    A program that accesses URLs.
+
+    Examples of such programs are cURL and wget.
+    Capabilities include:
+    -   Generating a command to access a URL via
+            :func:`~docker_cli.URLReader.generate_read_command`
+    -   Returning the name of the program.
+    """
+
+    @staticmethod
+    @abstractmethod
+    def generate_read_command(
+        target: str,
+        *,
+        output_target: Optional[Union[str, os.PathLike[str]]] = None,
+        stringify: bool = False,
+    ) -> Union[List[str], str]:
+        """
+        Returns a set of commands to read a URL.
+
+        Parameters
+        ----------
+        target : str
+            The target URL.
+        output_target : os.PathLike, optional
+            The file to redirect the URL contents to. If None, the URL contents
+            will go to stdout. Defaults to None.
+        stringify : bool, optional
+            Whether to return a string or list - if true, will return a string.
+            Defaults to False.
+
+        Returns
+        -------
+        Union[List[str], str]
+            The commands.
+        """
+        ...
+
+    @staticmethod
+    @abstractmethod
+    def get_name() -> str:
+        """
+        Returns the name of the command.
+
+        Returns
+        -------
+        str
+            The command name.
+        """
+        ...
+
+
+class Wget(URLReader):
+    @staticmethod
+    def generate_read_command(
+        target: str,
+        *,
+        output_target: Optional[Union[os.PathLike[str], str]] = None,
+        stringify: bool = False,
+    ) -> Union[List[str], str]:
+        retval = ["wget", target]
+        if output_target is not None:
+            retval += ["-O", str(output_target)]
+        if stringify:
+            return " ".join(retval)
+        return retval  # pragma: no cover
+
+    @staticmethod
+    def get_name() -> str:  # pragma: no cover
+        return "wget"
+
+
+class cURL(URLReader):
```

Comment:
Let's rename this class to `Curl` to be compliant with PEP8 naming conventions

```suggestion
class Curl(URLReader):
```